src/utils/delay.h

```diff
@@ -0,0 +1,62 @@
+/**
+ * @file delay.h
+ * @brief Adds functionality to use delays
+ *
+ * Adds methods to let the program wait for a certain amount of time. The time
+ * to wait can be specified in seconds, milliseconds or microseconds. Waiting a
+ * certain amount of microseconds is as precise as I would like but it comes
+ * very close. The deviation for milliseconds and seconds are less than 1% but
+ * are at least(!) the specified time.
+ *
+ * Note: The current implementation only works for a chip running on default
+ * speed (8MHz).
+ *
+ * Note: Interrupts can cause a delay to take longer than specified as this
+ * implementation uses busy waiting instead of a timer.
+ *
+ * \copyright Copyright 2017 /Dev. All rights reserved.
+ * \license This project is released under MIT license.
+ *
+ * @author Ferdi van der Werf <ferdi@slashdev.nl>
+ */
+
+#ifndef UTILS_DELAY_H
+#define UTILS_DELAY_H
+
+#include "compiler.h"
+
+OPTIMIZE_HIGH
+RAMFUNC
+extern void _cycle_delay(unsigned long n);
+
+#if F_CPU == 8000000UL
+#  define DELAY_MULT 1144
```

Comment:
Maybe you can add a calculation here that shows how you obtained DELAY_MULT, and show that the deviation indeed is smaller than 1%.

Reply:
I've added a comment regarding this define, can you review it?